### PR TITLE
[IOPAE-1601] mock receipt download

### PR DESCRIPTION
**Pull request by loricurreli**
#### List of Changes

Add the download function in the payment detail section

#### Motivation and Context

We need to download the file containing the payment's receipt

#### How Has This Been Tested?

There is a dedicated test for the new api /api/pagamenti/ricevuta/:id

#### Screenshots (if appropriate):

Simulator Screenshot - iPhone 15 Pro - 2024-12-03 at 14 35 28

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/ui/lib/components/icon/Icon.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { SvgIconProps } from "@mui/material/SvgIcon";
 
+import Download from "./svg/Download";
```
Comment (adelloste) [inline packages/ui/lib/components/icon/Icon.tsx:3]:
```suggestion
import IconDownload from "./svg/Download";
```

File: packages/core/lib/wrappers/with-jwt-auth-handler.ts [hunk 2]
```diff
@@ -9,17 +9,21 @@ export const withJWTAuthHandler =
...
-  (request) => {
+  (request, ctx) => {
     const { auth } = request;
 
     if (!auth) {
       return handleUnauthorizedErrorResponse("No Authentication provided");
     }
 
     return handler(request, {
+      params: ctx.params,
```
Comment (adelloste) [inline packages/core/lib/wrappers/with-jwt-auth-handler.ts:26]:
```suggestion
      ...ctx,
```

**Review by adelloste — CHANGES_REQUESTED** [review]

File: apps/payments/src/lib/bff/business.ts [hunk 3]
```diff
@@ -22,3 +22,28 @@ export const retrievePayments = async (
...
 };
+
+export const retrievePaymentReceipt = async (
+  additionalDataJWS: string,
+  token: string,
+  paymentRequestId: string,
+) => {
+  try {
+    return await getExternalApiClient().stampaRicevutaTelematica({
+      headers: {
```
Comment (giuseppedipinto) [inline apps/payments/src/lib/bff/business.ts:33]: we are forgetting `codiceFiscale` as header: it's required in all apis

File: apps/payments/src/pages/payments/[requestId].tsx [hunk 4]
```diff
@@ -50,7 +74,21 @@ export default function PaymentDetails() {
 
   return (
     <>
-      <Stack my={3} spacing={2}>
+      <Stack spacing={2}>
+        {data.descrizioneTipoPratica && (
+          <SectionTitle label={data.descrizioneTipoPratica} />
+        )}
+        <Box width="max-content">
+          <Button
```
Comment (giuseppedipinto) [inline apps/payments/src/pages/payments/[requestId].tsx:82]: let's check if the button should be displayed only under certain conditions _(example: particular practice status, even if we know that the practice status is not an enumeration...)_

**Review by giuseppedipinto — CHANGES_REQUESTED** [review]
I left a couple of comments that are meant to be some food for thought before merge

**Review by adelloste — APPROVED** [review]

**Review by giuseppedipinto — APPROVED** [review]

**Pull request merged**